base/src/test/java/io/spine/query/RecordQueryBuilderTest.java

```diff
@@ -154,22 +155,41 @@ void byEitherColumn() {
                             .either((r) -> r.where(isin).is(isinValue),
                                     (r) -> r.where(is_traded).is(stocksAreTraded))
                             .build();
-            ImmutableList<QueryPredicate<Manufacturer>> predicates = query.subject()
-                                                                          .predicates();
-            assertThat(predicates).hasSize(2);
-
-            QueryPredicate<Manufacturer> actualAnd = predicates.get(0);
-            assertThat(actualAnd.operator()).isEqualTo(AND);
-            List<SubjectParameter<Manufacturer, ?, ?>> andParams = actualAnd.parameters();
-            assertThat(andParams).hasSize(1);
-            assertHasParamValue(andParams, when_founded, LESS_THAN, THURSDAY);
-
-            QueryPredicate<Manufacturer> actualEither = predicates.get(1);
-            assertThat(actualEither.operator()).isEqualTo(OR);
-            List<SubjectParameter<Manufacturer, ?, ?>> orParams = actualEither.parameters();
-            assertThat(orParams).hasSize(2);
-            assertHasParamValue(orParams, isin, EQUALS, isinValue);
-            assertHasParamValue(orParams, is_traded, EQUALS, stocksAreTraded);
+            QueryPredicate<Manufacturer> rootPredicate = query.subject().predicate();
+            assertThat(rootPredicate.operator()).isEqualTo(AND);
+
+            ImmutableList<SubjectParameter<Manufacturer, ?, ?>> parameters =
+                    rootPredicate.parameters();
+            assertThat(parameters.size()).isEqualTo(1);
+            assertHasParamValue(parameters, when_founded, LESS_THAN, THURSDAY);
+
+            ImmutableList<QueryPredicate<Manufacturer>> children = rootPredicate.children();
+            assertThat(children.size()).isEqualTo(1);
+            QueryPredicate<Manufacturer> either = children.get(0);
+            assertThat(either.operator()).isEqualTo(OR);
+            assertThat(either.customParameters()).hasSize(0);
+            ImmutableList<SubjectParameter<Manufacturer, ?, ?>> params = either.parameters();
+            assertThat(params).hasSize(2);
+
+            assertHasParamValue(params, isin, EQUALS, isinValue);
+            assertHasParamValue(params, is_traded, EQUALS, stocksAreTraded);
```

Comment:
This test has several assertions. It looks like there are three aspects that we test. Can we have it as a nested class with sections of this method transformed into methods?

Reply:
@alexander-yevsyukov These are already nested classes, which test different aspects :) Besides, these assertions test basically a single thing, which is an exact number and types of resulting parameters joined with an expected operator. E.g. "We expect an outcome to be disjunction of two parameters with certain values" . Not sure how to split it.

Therefore, I would prefer not to make it deeper.